src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtPostcodeServiceTest.java

```diff
@@ -0,0 +1,148 @@
+    );
+    private static final List<String> POSTCODES_TO_BE_ADDED = Collections.singletonList(NEW_POSTCODE);
+    private static final int POSTCODE_COUNT = POSTCODES.size();
+
+    @Autowired
+    private AdminCourtPostcodeService adminService;
+
+    @MockBean
+    private CourtRepository courtRepository;
+
+    @MockBean
+    private CourtPostcodeRepository courtPostcodeRepository;
+
+    @Mock
+    private Court court;
+
+    private final List<CourtPostcode> courtPostcodes = Arrays.asList(
+        new CourtPostcode(TEST_POSTCODE1, court),
+        new CourtPostcode(TEST_POSTCODE2, court),
+        new CourtPostcode(TEST_POSTCODE3, court),
+        new CourtPostcode(TEST_POSTCODE4, court)
+    );
+
+    @Test
+    void shouldReturnAllCourtPostcodes() {
+        when(court.getCourtPostcodes()).thenReturn(courtPostcodes);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+
+        assertThat(adminService.getCourtPostcodesBySlug(COURT_SLUG))
+            .hasSize(POSTCODE_COUNT)
+            .containsExactlyElementsOf(POSTCODES);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenRetrievingPostcodesForNonExistentCourt() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> adminService.getCourtPostcodesBySlug(COURT_SLUG))
+            .isInstanceOf(NotFoundException.class)
+            .hasMessage(NOT_FOUND + COURT_SLUG);
+
+        verifyNoInteractions(courtPostcodeRepository);
+    }
+
+    @Test
+    void shouldAddCourtPostcodes() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(courtPostcodeRepository.save(any())).thenReturn(new CourtPostcode(NEW_POSTCODE, court));
+
+        assertThat(adminService.addCourtPostcodes(COURT_SLUG, POSTCODES_TO_BE_ADDED))
+            .hasSize(1)
+            .containsExactly(NEW_POSTCODE);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenAddingPostcodesForNonExistentCourt() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> adminService.addCourtPostcodes(COURT_SLUG, POSTCODES))
+            .isInstanceOf(NotFoundException.class)
+            .hasMessage(NOT_FOUND + COURT_SLUG);
+
+        verifyNoInteractions(courtPostcodeRepository);
+    }
+
+    @Test
+    void shouldDeleteCourtPostcodes() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(court.getId()).thenReturn(TEST_COURT_ID);
+        when(courtPostcodeRepository.findByCourtIdAndPostcode(TEST_COURT_ID, TEST_POSTCODE2)).thenReturn(Optional.of(courtPostcodes.get(1)));
+        when(courtPostcodeRepository.findByCourtIdAndPostcode(TEST_COURT_ID, TEST_POSTCODE3)).thenReturn(Optional.of(courtPostcodes.get(2)));
+
+        adminService.deleteCourtPostcodes(COURT_SLUG, POSTCODES_TO_BE_DELETED);
+        verify(courtPostcodeRepository).deleteAll(any());
```

Comment:
Is there a way to assert that the three above are being deleted? Any doesn't really catch if we are sending through the same one (for example).

Reply:
I think this is probably best to be tested in the functional test as they can call the GET endpoint and check what is remaining there